### DISCUSSION TITLE
typescript 编译器报告 @ant-design/pro-components xxx 仅在大小写方面与包含的文件名xxx 环境信息:  @umijs/max: ^4.0.71 typescript: 5.1.3 node: v16.9.1

### DISCUSSION THREAD

**Discussion by Jikerr** [umijs/umi#11317] [General]
typescript 编译器报告 @ant-design/pro-components xxx 仅在大小写方面与包含的文件名xxx
环境信息: 
@umijs/max: ^4.0.71
typescript: 5.1.3
node: v16.9.1
![image](https://github.com/umijs/umi/assets/29561059/fffd4ce6-2b4d-424a-9e60-f245477b3f89)

**Comment by fz6m** [comment]
删掉 `node_modules` ，重装一下依赖，然后升级下 vscode 版本看看。

如有问题，需要提供一个视频的最小复现，这个问题已经修复过了，如还有问题，需要提供更详细的上下文信息才能帮助排查。

**Reply by Jikerr** [reply]
很高兴收到您的回复，我按照你给出的方案
1.删除了node_modules文件夹
2.重装了依赖（npm install
3.我检查了我的vs code版本，目前在1.79.0，无法更新到更新的版本了

我的问题依然没有得到解决，我继续补充相关环境信息给您，希望能得到你的帮助。

OS: windows 10
Nodejs:16.17.1
Typescript:5.1.3
 
我使用了umi cli创建了工程，执行：
npx create-umi@latest
在选择 app template步骤时，我选择了ant design pro
继而执行完自动化的步骤后，我得到了自动生成的示例代码，在位置 src/page/Home/index.tsx中的第三行导入语句，import {PageContainer} form "@ant-design/pro-components";我的编译器报告了问题，详细信息如下：

[{
	"resource": "/d:/J2EE/workspace_me/myapp2/src/pages/Home/index.tsx",
	"owner": "typescript",
	"code": "1149",
	"severity": 8,
	"message": "文件名“d:/J2EE/workspace_me/myapp2/node_modules/@ant-design/pro-components/es/index.d.ts”仅在大小写方面与包含的文件名“d:/j2ee/workspace_me/myapp2/node_modules/@ant-design/pro-components/es/index.d.ts”不同。\n  程序包含该文件是因为:\n    通过 \"D:/J2EE/workspace_me/myapp2/node_modules/@ant-design/pro-components\" 从具有 packageId \"@ant-design/pro-components/es/index.d.ts@2.6.2\" 的文件 \"D:/J2EE/workspace_me/myapp2/src/.umi/plugin-layout/types.d.ts\" 引用了库类型……

如上是我的复现方法，我按照上述步骤重新新建了一个工程问题依旧。

再次感谢您能为我提供指导。

**Reply by fz6m** [reply]
npm 版本是多少，更新下 npm 版本看下，或者尝试用 yarn 、pnpm 看看。

**Reply by Jikerr** [reply]
问题已解决，使用pnpm后正常。

**Reply by Jikerr** [reply]
> npm 版本是多少，更新下 npm 版本看下，或者尝试用 yarn 、pnpm 看看。

感谢ヾ(≧∇≦谢谢≧∇≦)ノ
